### DISCUSSION TITLE
有关使用 Docker 搭建时的反代问题

### DISCUSSION THREAD

**Discussion by Robotxm** [AlistGo/alist#633] [Q&A] [answered]
我起了个 Docker 用来搭建 Alist，为了支持 HTTPS，同时使用了 Nginx 进行反代。设置如下：

```
server_name example.com;
location /disk {
	proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
	proxy_set_header Host $http_host;
	proxy_set_header X-Real-IP $remote_addr;
	proxy_redirect off;
	proxy_pass http://127.0.0.1:5244;
	# Max file size to upload
	client_max_body_size 20000m;
}
```

但是 Alist 的前端仍然会请求 https://example.com/api/public/settings 而不是 https://example.com/disk/api/public/settings。

请问该如何设置呢？

**Comment by xhofe** [comment, marked as answer]
理论上来说不会 发一下地址，不方便公开可以通过：[邮件](mailto:support@nn.ci)

**Reply by Robotxm** [reply]
https://myxdu.moefactory.com/disk

感谢。

**Reply by xhofe** [reply]
暂不支持直接反代到二级目录，建议使用二级域名。如果非得这么做，两种方法：
1. 修改前端的源代码重新编译
2. 反代多个目录，大致有`/d`, `/p`, `/i`, `/api`，但是这样做可能会和你的一级网站产生冲突。

**Reply by Robotxm** [reply]
那我先用二级域名凑合一下。

希望以后能支持反代到二级目录。感谢解答。
